tests/test_statistics.py

```diff
@@ -601,10 +601,12 @@ def test_bincount_minlength(self, array, minlength):
     )
     def test_bincount_weights(self, array, weights):
```

Comment:
Please move that tests to `tests/test_histogram.py::TestBincount` to have them in one place. Or remove if they duplicate the scenario you already covered there.
In any case `allow_fall_back_on_numpy` fixture has to be removed.